doc: automatic image optimisation detection (#11710)
supabase/supabase: apps/docs/pages/guides/storage/image-transformations.mdx

})
```

> ## Automatic Image Optimisation Detection (WebP)

Reviewer: Thoughts on just calling this "Automatic Image Optimization?" "Detection" seems a bit redundant.

When using the image transformation API we will automatically find the best format supported by the browser and return that to the client, without any code change.
For instance, if you use Chrome when viewing a jpeg image and using transformation options, you'll see that the content-type returned is `webp`.

> In result, this will lower the bandwidth that you send to your users and your application will load much faster.

Reviewer 2: ```suggestion
As a result, this will lower the bandwidth that you send to your users and your application will load much faster.
```

<Admonition type="note">
    We currently only support WebP. AVIF support will come in the near future
</Admonition>

**Disabling automatic optimisation:**

In case you'd like to return the original format of the image and **opt-out** from the automatic image optimization detection, you can pass the `format=origin` parameter when requesting a transformed image,

> this is also supported in the SDK starting from v2.2.0

Reviewer 2: ```suggestion
this is also supported in the JavaScript SDK starting from v2.2.0
```

Reviewer 2: Note to @inian to update the docs. cc @dshukertjr for updating the dart lib with this option

| BMP    | `bmp`     | ☑️     | ☑️     |
| TIFF   | `tiff`    | ☑️     | ☑️     |

## Self Hosting

Our solution to image resizing and optimisation can be self-hosted as with any other Supabase product.
Under the hood we use the awesome [Imgproxy](https://imgproxy.net/)

> #### Imgproxy Configuration:

Reviewer 2: This guide is for the hosted platform. Can we move the imgproxy configuration to the [self hosting section](https://supabase.com/docs/guides/self-hosting/storage/config) instead?

Author: I think i can add the configuration in the section you mentioned, but i think it would still be nice to showcase how simple could be to self host it

Simply deploy an imgproxy container with the following configuration:

```yaml
imgproxy:
  image: darthsim/imgproxy
  environment:
    - IMGPROXY_ENABLE_WEBP_DETECTION=true
    - IMGPROXY_JPEG_PROGRESSIVE=true
```

Note: make sure that this service can only be reachable within an internal network and not exposed to the public internet

> #### Storage API Configuration:
> 
> Once [Imgproxy](https://imgproxy.net/) is deployed we need to configure a couple of environment variables in your self-hosted [storage-api](https://github.com/supabase/storage-api) service as follow:

Reviewer: ```suggestion
### Storage API Configuration

After [Imgproxy](https://imgproxy.net/) is deployed, configure the environment variables in your self-hosted [storage-api](https://github.com/supabase/storage-api) service:
```
